Shared/analysis/LineOfSightController.cpp

```diff
@@ -391,98 +403,129 @@ void LineOfSightController::selectOverlayIndex(int selectOverlayIndex)
+      setupNewLineOfSight(lineOfSight);
+    }
+  }
+}
 
-      m_visibleByConnections.append(connect(lineOfSight, &GeoElementLineOfSight::targetVisibilityChanged, this, [this]()
-      {
-        int visibleCount = 0;
-        auto* losList = m_lineOfSightOverlay->analyses();
-        const int count = losList->rowCount();
-        for (int i = 0; i < count; ++i)
-        {
-          auto* analysis = losList->at(i);
-          if (!analysis)
-            continue;
+bool LineOfSightController::resetAnalysis(size_t featureCount)
+{
+  constexpr int maxFeatures = 32; // Due to performance reasons, limit the number of features which can be used in the analysis
+  if (featureCount > maxFeatures)
+  {
+    emit toolErrorOccurred(QString("There are too many points in this layer (%1).\n Please choose another one with %2 or fewer points.")
+                           .arg(QString::number(featureCount), QString::number(maxFeatures)),
+                           QStringLiteral("For performance reasons, Line of Sight analysis is limited to a maximum number of features"));
+    emit selectOverlayFailed();
+    return false;
+  }
+
+  if (!m_locationGeoElement)
+    getLocationGeoElement();
+
+  if (!m_locationGeoElement)
+  {
+    emit toolErrorOccurred(QStringLiteral("Failed to get location"), QStringLiteral("Unable to find My Location GeoElement for GeoElementLineOfSight"));
+    return false;
+  }
 
-          auto* lineOfSight = qobject_cast<GeoElementLineOfSight*>(analysis);
-          if (!lineOfSight)
-            continue;
+  for (const auto& conn : m_visibleByConnections)
+    disconnect(conn);
 
-          // don't count Line of Sight from the current location
-          if (lineOfSight->observerGeoElement() == m_locationGeoElement)
-            continue;
+  m_visibleByConnections.clear();
+  setVisibleByCount(0);
 
-          if (lineOfSight->targetVisibility() == LineOfSightTargetVisibility::Visible)
-            visibleCount += 1;
-        }
+  // clear the QObject used as a parent for Line of Sight results
+  if (m_lineOfSightParent)
+  {
+    delete m_lineOfSightParent;
+    m_lineOfSightParent = nullptr;
+  }
+  m_lineOfSightParent = new QObject(this);
+  return true;
+}
 
-        setVisibleByCount(visibleCount);
-      }));
-    }
+void LineOfSightController::setupNewLineOfSight(GeoElementLineOfSight* lineOfSight)
+{
+  lineOfSight->setVisible(m_analysisVisible);
+  m_lineOfSightOverlay->analyses()->append(lineOfSight);
 
-    // if no early returns or failures happened pass success=true along to the continuation
-    return true;
-  }).then(this, [this](bool success)
+  m_visibleByConnections.append(connect(lineOfSight, &GeoElementLineOfSight::targetVisibilityChanged, this, [this]()
   {
-    if (!success)
-      emit selectOverlayFailed();
-  });
+    int visibleCount = 0;
+    const auto* losList = m_lineOfSightOverlay->analyses();
+    for (auto losIt = losList->begin(); losIt < losList->end(); losIt++)
```

Comment:
```suggestion
    for (auto losIt = std::being(losList); losIt != std::end(losList); ++losIt)
```

I don't think using less than comparisons with iterators is typical.

Reply:
Switching to ranged loop here as well.